### PR TITLE
CEDS-1479 Create Endpoint to persist to the backend

### PR DESCRIPTION
**Pull request by ed0906**
A lightweight endpoint to Persist `ExportsDeclaration`s to the back end

### CODE REVIEW

File: app/uk/gov/hmrc/exports/controllers/DeclarationController.scala [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+import scala.concurrent.ExecutionContext.Implicits.global
+
+@Singleton
+class DeclarationController @Inject()(
+  declarationService: DeclarationService,
+  override val controllerComponents: ControllerComponents
+) extends RESTController {
+
+  def post(): Action[ExportsDeclaration] = Action.async(parsingJson[ExportsDeclaration]) { implicit request =>
```
Comment (ed0906) [inline app/uk/gov/hmrc/exports/controllers/DeclarationController.scala]: Do we think this needs any kind of Auth?

**Comment by ed0906** [comment]
I've done this as a completely new slice (new Controller+Service+Repository) as I suspect a lot of the old stuff will end up getting replaced

File: app/uk/gov/hmrc/exports/controllers/DeclarationController.scala (hunk 1 above)
Comment (bambuchaAdm) [reply app/uk/gov/hmrc/exports/controllers/DeclarationController.scala]: I think so - it even describe in platform patterns - [Assumed Upstream Security](https://confluence.tools.tax.service.gov.uk/display/AR/Assumed+Upstream+Security)

Comment (ed0906) [reply app/uk/gov/hmrc/exports/controllers/DeclarationController.scala]: Auth Added

**Pull request merged**